### PR TITLE
Add trackOutlineColor for Switch and SwitchListTile

### LINKED ISSUE #119334: Add "inactiveBorderColor" to Material 3 Switch
## Use case

Like:

```dart
Switch(
  value: dark,
  inactiveBorderColor: Colors.white
  onChanged: (bool value) {
    setState(() {
      dark = value;
    });
  },
),
```

## Proposal

Now i can't change the border color😔

微信截图_20230127225414

### PR DESCRIPTION
**Pull request by QuncCccccc**
Fixes #119334

This PR is to add `trackOutlineColor` for `Switch` and `SwitchListTile`.

Screenshot 2023-02-06 at 4 14 17 PM

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/material/switch.dart [hunk 1]
```diff
@@ -333,6 +335,40 @@ class Switch extends StatelessWidget {
...
+  ///
+  /// ```dart
+  /// Switch(
+  ///   value: true,
+  ///   onChanged: (_) => true,
+  ///   trackOutlineColor: MaterialStateProperty.resolveWith<Color>((Set<MaterialState> states) {
+  ///     if (states.contains(MaterialState.disabled)) {
+  ///       return Colors.orange.withOpacity(.48);
+  ///     }
+  ///     return Colors.orange;
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/switch.dart:360]: If you returned null here, like:
```dart
return null; // Use the default color
```
That would remind developers what returning null from a MaterialStateProperty.resolve() method means.

File: packages/flutter/test/material/switch_list_tile_test.dart [hunk 2]
```diff
@@ -618,4 +618,117 @@ void main() {
...
+      }
+      return inactiveEnabledTrackOutlineColor;
+    }
+
+    final MaterialStateProperty<Color> trackOutlineColor = MaterialStateColor.resolveWith(getOutlineColor);
+
+    Widget buildSwitchListTile({required bool enabled, required bool selected}) {
+      return wrap(
+        child: StatefulBuilder(
+            builder: (BuildContext context, StateSetter setState) {
```
Comment (HansMuller) [inline packages/flutter/test/material/switch_list_tile_test.dart:646]: Do we really need a StatefulBuilder here? we do't appear to call `setState`

**Review by HansMuller — APPROVED** [review]
NICE. LGTM

File: packages/flutter/lib/src/material/switch.dart (hunk 1 above)
Comment (QuncCccccc) [reply packages/flutter/lib/src/material/switch.dart]: I see. Just changed. Thanks!

File: packages/flutter/test/material/switch_list_tile_test.dart (hunk 2 above)
Comment (QuncCccccc) [reply packages/flutter/test/material/switch_list_tile_test.dart:646]: Just removed all the unnecessary `StatefulBuilder`s in test.

**Comment by QuncCccccc** [comment]
> is already in stable channel?

Hi! No, it's not in stable yet. The change will go to stable on next Flutter stable increment.

**Comment by augustineonu** [comment]
Great. 
I have an issue making use of the material 3. the default switch before material 3 has an android kinda look. where the track is slim, whihc coulde be customized to be fat. but now the track defaults to the fat tracker. and i need the kinda slim one. i can describe it as the default android switch. 
I hae aslo added an image of what i am trying to achieve 
Screenshot 2023-11-13 191930

**Comment by augustineonu** [comment]
okay i acived by setting  'useMaterial3: false,'

**Pull request merged**